### PR TITLE
Some patches for qip.Gate

### PR DESCRIPTION
**Pull request by BoxiLi**
Some small changes:

1. remove repetitive assignments of `self.arg_value` and` self.arg_label`
2. move the description to the definition of class so the document will show it
3. raise the corresponding error if targets is None while required or not an integer, e.g `Gate("SWAP")` or `Gate("SWAP", [0.5, 2])`

### CODE REVIEW

File: qutip/qip/circuit.py [hunk 1]
```diff
@@ -31,9 +31,11 @@
 #    OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ###############################################################################
 
-import numpy as np
+from collections.abc import Iterable
```
Comment (nathanshammah) [inline qutip/qip/circuit.py]: Can you update this import according to #1003 ?

File: qutip/qip/circuit.py [hunk 2]
```diff
@@ -45,46 +47,50 @@ class Gate(object):
     """
     Representation of a quantum gate, with its required parametrs, and target
     and control qubits.
+    
+    Parameters
+    ----------
+    name : String
```
Comment (nathanshammah) [inline qutip/qip/circuit.py]: lower capital unless this is a class. String -> string. Same applies for other parameters belows.

File: qutip/qip/circuit.py [hunk 3]
```diff
@@ -45,46 +47,50 @@ class Gate(object):
...
+        Gate name.
+    targets : List or Integer
+        Gate targets.
+    controls : List or Integer
```
Comment (nathanshammah) [inline qutip/qip/circuit.py]: Integer might not be defining an object (but I might be wrong). Maybe `int` is more appropriate? But maybe also with a float that is an integer, the function works.

File: qutip/qip/circuit.py [hunk 4]
```diff
@@ -45,46 +47,50 @@ class Gate(object):
...
+    arg_value : Float
+        Argument value(phi).
+    arg_label : String
+        Label for gate representation.
     """
 
     def __init__(self, name, targets=None, controls=None, arg_value=None,
                  arg_label=None):
         """
         Creates a gate with specified parameters.
```
Comment (nathanshammah) [inline qutip/qip/circuit.py]: Create a gate (etc).

File: qutip/qip/circuit.py (hunk 1 above)
Comment (BoxiLi) [reply qutip/qip/circuit.py]: I'm a bit confused. In my python version 3.7.3, this warning occurs when I run
`from collections import Iterable`
but doesn't when I run
`from collections.abc import Iterable`
It seems to me that the second one is the up-to-date one?

Comment (BoxiLi) [reply qutip/qip/circuit.py]: @nathanshammah  Thanks for the suggestions. I have corrected all other doc formats. I think the importation I used here `from collections.abc import Iterable` should be the up-to-date one.

**Pull request merged**